src/Behat/Context/CookieComplianceContext.php

```diff
@@ -113,4 +169,157 @@ public function iWaitCookieBannerAppears() {
+    }
+    catch (\Exception $exception) {
+      $errors[] = sprintf('  - %s', $exception->getMessage());
+    }
+
+    try {
+      $this->analyzeThirdPartyCookies();
+    }
+    catch (\Exception $exception) {
+      $errors[] = sprintf('  - %s', $exception->getMessage());
+    }
+
+    if (!empty($errors)) {
+      throw new \Exception(sprintf("Errors found:\n%s", implode("\n", $errors)));
+    }
+
+  }
+
+  /**
+   * Analyze there aren't cookies at the host.
+   *
+   * If there are, it will throw an exception
+   * and show a table with all the loaded cookies.
+   */
+  protected function analyzeHostCookies() {
+    $webdriver_session = $this->getSession()->getDriver()->getWebDriverSession();
+    $cookies = array_filter($webdriver_session->getAllCookies(), function ($cookie) {
+      return !in_array($cookie['name'], $this->cookiesIgnored);
+    });
+
+    $cookies_ignored_loaded = array_intersect($this->cookiesIgnored, array_column($cookies, 'name'));
+    if (!empty($cookies_ignored_loaded)) {
+      print sprintf('The following cookies are loaded but they are setup as ignored: %s', implode(',', $cookies_ignored_loaded));
+    }
+
+    if (!empty($cookies)) {
+      $this->showCookies($cookies);
+      throw new \Exception('There are cookies but there should not be any cookie.');
+    }
+  }
+
+  /**
+   * Shows the list of cookies that have been found.
+   *
+   * Array of cookies.
+   *
+   * @param array $cookies
+   *   List of cookies to print.
+   */
+  protected function showCookies(array $cookies) {
+    $output = new BufferedOutput();
+    $cookiesTable = new Table($output);
+
+    $cookiesTable->setHeaders(['Domain', 'Name', 'Value']);
+
+    $cookiesTable->setRows(array_map(function ($cookie) {
+      return [
+        $cookie['domain'],
+        $cookie['name'],
+        $cookie['value'],
+      ];
+    }, $cookies));
+
+    $output->writeln("\nCookies found:\n");
+
+    $cookiesTable->render();
+
+    echo $output->fetch();
+  }
+
+  /**
+   * Analyze there aren't third party cookie iframes at the host.
+   *
+   * If there are, it will throw an exception and show a table with the
+   * list of iframes that are loading cookies.
+   */
+  protected function analyzeThirdPartyCookies() {
+    $page_html_document = new \DOMDocument('4.0');
+
+    $page_html_document->loadHTML($this->getSession()->getPage()->getOuterHtml(), LIBXML_NOERROR);
+    $page_html_xpath = new \DOMXPath($page_html_document);
+
+    $third_party_cookie_urls_loaded = [];
+
+    /** @var \DOMElement $iframe */
+    foreach ($page_html_xpath->query('//iframe') as $iframe) {
+      $iframe_src = $iframe->getAttribute('src');
+      if (!empty($iframe_src)) {
+        $iframe_host = parse_url($iframe_src, PHP_URL_HOST);
+        foreach (static::THIRD_PARTY_COOKIE_HOSTS as $third_party_cookie_host) {
+          if (!in_array($third_party_cookie_host, $this->cookiesThirdPartyDomainsIgnored) && str_ends_with($iframe_host, $third_party_cookie_host)) {
```

Comment:
I don't know if I'm understanding this part correctly, but we are only supposed to check the domains that are inside the THIRD_PARTY_COOKIE_HOSTS constant and also have been passed in cookiesThirdPartyDomainsIgnored? Shouldn't they be complementary lists? I mean, check the domains that are in the constant and also the ones that are passed in the configuration.